docs/src/_contribute/cli.md

```diff
@@ -0,0 +1,255 @@
+  For example, the `meltano schedule` feature-group contains commands related to managing Meltano schedules.
+- command/sub_command are used to perform specific tasks or used to group a set of command around a specific feature
+  or task. Sub-commands take the form of parameters, but require additional arguments or options themselves.
+- Arguments are positional parameters that are passed to a command. Arguments do not require additional options or
+  arguments - otherwise they would be considered a sub-command.
+- Options (switch, option flags, or flags) are options that alter the behavior (e.g. `--dry-run/--verbose`) or named
+  input options (e.g. `--tasks=`). They come in two forms. A long form: `--option-name`, and a short form: `-o`.
+
+Given a command like `meltano job set JOB_NAME --tasks=[<task>...]`:
+
+- `meltano` is the global click command group for Meltano. This is the main entry point to the CLI. When we refer to global options this is the level we refer to.
+- `job` is the click command group for the `job` command. This is the main command group for the `job` command. This is typically the top level for a Meltano feature.
+- `set` is a `sub-command` of the `job` command.
+- `JOB_NAME` is an `argument` of the `set` sub-command.
+- `--tasks` is named `option` of the `set` sub-command.
+
+For a technical explanation of how commands and groups work see https://click.palletsprojects.com/en/8.0.x/commands/
+
+### Use of short and long options
+
+The options you expect to be used most frequently should always include both the long AND short variation of the option.
+For example `--help` and `-h`, `--tasks` and `-t`, `--force` and `-f`.
+
+
+### Desired verb/command linkage and structure
+
+1. Organize in a command group around a feature below the top level where possible.
+2. Coalesce commands that operate on the same resource of a feature as a group of sub-commands.
+3. Use transitive verbs like `remove`, `set`, `list`, `get` as sub-commands to perform work within a group where appropriate.
+
+```
+meltano <feature-group> add <something>
+meltano <feature-group> delete <something>
+meltano <feature-group> get <something>
+```
+
+#### Standardized verbs
+
+- Use `add` and `remove` to create or remove a construct like "schedule" or "job"
+- Use `set` to override or update a construct after creation as in `meltano jobs JOB_NAME set`
+- Use `get`, `set`, `list`, `clear` for operating on variable like artifacts like "state" and "secrets"
+- Use `list` to enumerate items as in `meltano job list`
+- Use `describe` to show details about an item
+
+### Abbreviations
+
+For excessively long options, especially in cases where no short flag is provided or a commonly used abbreviation is exists,
+you should allow the use of the abbreviation as a documented alias:
+
+```
+meltano --env=production
+meltano --environment=production
+```
```

Comment:
Ditto comments from above. Sometime abbreviations can cause confusion and I would hate to (1) slow down an otherwise-ready PR to discuss whether an abbreviation is necessary + sufficiently helpful, or (2) say "yes" too quickly because we want the feature to go out, without thinking through if the abbreviation is needed.

I think we can use the above thread for this discussion but just wanted to call it out here inline as well.

Perhaps not to others, but as a case in point, `--env` brings to mind a specific env variable mapping, and is less obviously a declaration of the environment name. While I realize `--environment` is longer to type, I just don't want to rush through these decisions without thinking through them.

Reply:
Personally, I think having only long options is terrible ux. Everytime i have to type `--environment` I get sad.